cmd/metrics-proxy/main.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"flag"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/canonical/metrics-k8s-proxy/internal/handlers"
+	"github.com/canonical/metrics-k8s-proxy/internal/k8s"
+	"github.com/canonical/metrics-k8s-proxy/internal/util"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/gorilla/mux"
+)
+
+// Parses the label selector and timeout from command line arguments.
+func parseFlags() (map[string]string, time.Duration) {
+	labelSelector := flag.String("labels", "", "Label selector for watching pods (e.g., 'app=ztunnel')")
+	timeout := flag.Duration("timeout", 9*time.Second, "HTTP server read and write timeout (e.g., 15s, 1m)")
```

Comment:
This could be clearer.  "timeout" sounds like we'll return a failure if this process takes too long.  I don't think someone reading this will understand that the timeout is really the cutoff on any individual scrape duration.  Maybe "scrape_timeout" and "Maximum allowed time for any given scrape" or something like that?